crypto/ec/ec_backend.c

```diff
@@ -53,6 +48,15 @@ int ec_set_param_ecdh_cofactor_mode(EC_KEY *ec, const OSSL_PARAM *p)
     return 1;
 }
 
+int ec_set_param_ecdh_cofactor_mode(EC_KEY *ec, const OSSL_PARAM *p)
+{
+    int mode;
+
+    if (!OSSL_PARAM_get_int(p, &mode))
+        return 0;
+    return ec_set_ecdh_cofactor_mode(ec, mode);
+}
+
```

Comment:
Instead of adding this small thing, may I suggest this?

``` diff
diff --git a/crypto/ec/ec_backend.c b/crypto/ec/ec_backend.c
index 98dd0ecf5d..73dd4d5903 100644
--- a/crypto/ec/ec_backend.c
+++ b/crypto/ec/ec_backend.c
@@ -224,8 +220,12 @@ int ec_key_otherparams_fromdata(EC_KEY *ec, const OSSL_PARAM params[])
         return 0;
 
     p = OSSL_PARAM_locate_const(params, OSSL_PKEY_PARAM_USE_COFACTOR_ECDH);
-    if (p != NULL && !ec_set_param_ecdh_cofactor_mode(ec, p))
-        return 0;
+    if (p != NULL) {
+        int mode;
 
+        if (!OSSL_PARAM_get_int(p, &mode)
+            || !ec_set_param_ecdh_cofactor_mode(ec, p))
+            return 0;
+    }
     return 1;
 }
diff --git a/providers/implementations/keymgmt/ec_kmgmt.c b/providers/implementations/keymgmt/ec_kmgmt.c
index a48b279547..bc66c7c1e4 100644
--- a/providers/implementations/keymgmt/ec_kmgmt.c
+++ b/providers/implementations/keymgmt/ec_kmgmt.c
@@ -542,14 +542,7 @@ const OSSL_PARAM *ec_settable_params(void)
 static
 int ec_set_params(void *key, const OSSL_PARAM params[])
 {
-    EC_KEY *eck = key;
-    const OSSL_PARAM *p;
-
-    p = OSSL_PARAM_locate_const(params, OSSL_PKEY_PARAM_USE_COFACTOR_ECDH);
-    if (p != NULL && !ec_set_param_ecdh_cofactor_mode(eck, p))
-        return 0;
-
-    return 1;
+    return ec_key_otherparams_fromdata(key, params);
 }
 
 static
```
```